### PR TITLE
Support encrypted welcomes

### PR DESCRIPTION
**Pull request by neekolas**
## Summary

- Implements https://github.com/xmtp/libxmtp/issues/423
- Depends on https://github.com/xmtp/xmtp-node-go/pull/337 and https://github.com/xmtp/proto/pull/137
- Adds support for encrypting welcome messages using the `hpke_public_key`

### CODE REVIEW

File: dev/docker/docker-compose.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 services:
   node:
-    image: xmtp/node-go:mls
+    image: xmtp/node-go:mls-dev
```
Comment (neekolas) [inline dev/docker/docker-compose.yml:3]: Revert this before merging

File: dev/gen_protos.sh [hunk 2]
```diff
@@ -6,7 +6,7 @@ if ! cargo install --list | grep "protoc-gen-prost-crate" > /dev/null; then
     fi
 fi
 
-if ! buf generate https://github.com/xmtp/proto.git#branch=main,subdir=proto; then
+if ! buf generate https://github.com/xmtp/proto.git#branch=nm/post-commit-action-updates,subdir=proto; then
```
Comment (neekolas) [inline dev/gen_protos.sh:9]: Revert this before merging

**Review by tuddman — APPROVED** [review]
While I'm not yet fully confident in understanding all of the subtleties in correctly implementing the welcome message encryption - there's nothing glaring that I can see.  But I gladly differ to others to give this a more thorough read-through before merging.

Without bike-shedding hard about `installation` V `installation_id` V `installation_key` - can you add a note explaining the decision / implications of going with one term vs. the other?

**Comment by neekolas** [comment]
> Without bike-shedding hard about installation V installation_id V installation_key - can you add a note explaining the decision / implications of going with one term vs. the other?

I've switched anything new in this PR to use `installation_key`. There are still a bunch of places in `libxmtp` that reference `installation_id`. But now there are none over the wire at least.

The identifier for an installation is both an `id` (it's unique) and is a public key. So, there's a case to be made for calling it either `installation_id` or `installation_key`. I guess `installation_pub_key` was also an option. But the decision has been made, we're going with `installation_key`. There'll need to be more PRs to update references in other places.

**Pull request merged**